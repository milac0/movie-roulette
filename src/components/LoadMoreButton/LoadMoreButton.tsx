import React, { useContext } from "react";
import { DataContext } from "../../context/DataContext";

interface Props {}

const LoadMoreButton: React.FC<Props> = () => {
  const { page, setPage } = useContext(DataContext);
  const handleClick = () => setPage(page + 1);

  return <button onClick={handleClick}>load more...</button>;
};

export default LoadMoreButton;
