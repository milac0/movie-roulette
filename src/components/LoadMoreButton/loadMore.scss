@import "../../theme/variables";

.loadMore {
  display: flex;
  justify-content: center;
  margin-bottom: 3em;

  button {
    background: $primary;
    border: none;
    border-radius: 5px;
    color: white;
    font-size: 1rem;
    outline: none;
    padding: 1em;
    width: 5em;

    &:hover {
      background: $primaryDark;
      cursor: pointer;
    }
  }
}

@media (min-width: 414px) {
  .loadMore {
    button {
      width: 10em;
    }
  }
}
