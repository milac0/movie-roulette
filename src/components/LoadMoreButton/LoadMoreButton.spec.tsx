import React from 'react';
import renderer from 'react-test-renderer';

import LoadMoreButton from './LoadMoreButton';

describe('LoadMoreButton renders correctly', () => {
    test('snapshot renders', () => {
        const tree = renderer.create(<LoadMoreButton />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

