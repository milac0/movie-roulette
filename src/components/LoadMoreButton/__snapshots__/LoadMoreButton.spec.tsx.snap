// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadMoreButton renders correctly snapshot renders 1`] = `
<div
  className="loadMore"
>
  <button
    onClick={[Function]}
  >
    load
  </button>
</div>
`;
